openstack_dashboard/test/selenium/integration/test_user_credentials.py

```
@@ -0,0 +1,175 @@
+    user_credential_form = driver.find_element_by_css_selector(
+        ".modal-dialog form")
+    widgets.select_from_specific_dropdown_in_form(
+        user_credential_form, 'id_cred_type', 'EC2')
+    user_credential_form.find_element_by_id("id_data").clear()
+    user_credential_form.find_element_by_id("id_data").send_keys(
+        '{"access": "acs", "secret": "scrt"}')
+    widgets.select_from_specific_dropdown_in_form(
+        user_credential_form, 'id_project', 'admin')
+    user_credential_form.find_element_by_css_selector(".btn-primary").click()
+    messages = widgets.get_and_dismiss_messages(driver)
+    assert f'Success: User credential created successfully.' in messages
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)